hlua/src/lib.rs

```diff
@@ -224,6 +224,20 @@ pub unsafe trait AsMutLua<'lua>: AsLua<'lua> {
 // TODO: probably no longer necessary
 #[derive(Copy, Clone)]
 pub struct LuaContext(*mut ffi::lua_State);
+
+impl LuaContext {
+    /// Return a pointer to the inner `lua_State` for this context. This is an escape hatch that
+    /// lets the caller perform arbitrary operations against the FFI directly.
+    ///
+    /// Be careful: performing operations on this state might invalidate assumptions made in
+    /// higher-level APIs. For example, pushing a value onto the Lua stack will cause `PushGuard`s
+    /// in Rust code to be out of sync with the Lua stack.
+    #[inline]
+    pub fn state_ptr(&self) -> *mut ffi::lua_State {
```

Comment:
This shouldn't return a `*mut ffi::lua_State`, otherwise people are going to run into semver issues.
Instead I think it should return a `*mut c_void`, where `c_void` is imported from `std::os::raw::c_void`.

Reply:
Hmm, by "semver issues" do you mean: say `hlua` moves to `lua52-sys 0.2`, but the user's using `lua52-sys 0.1`?

If so then I think this is still preferable to returning a `void*`. For a `void*`, to do anything useful a cast is always necessary. For a `ffi::lua_State` pointer a cast is only necessary if the version has changed.

(I might be misunderstanding what you're saying -- in that case feel free to correct me!)

Reply:
Per some discussion here, an alternate solution would be to export the entire `ffi` module. That way callers that want to use the version of `lua52-sys` that corresponds to the `hlua` in use can just use it through `hlua::ffi`.

What do you think?